app/models/portfolio_item.rb

```diff
@@ -18,4 +18,13 @@ class PortfolioItem < ApplicationRecord
 
   belongs_to :portfolio
   validates :service_offering_ref, :presence => true
+
+  def item_workflow_ref
+    hierarchy_search = Catalog::HierarchicalModelSearch.new("workflow_ref", id, %w[PortfolioItem Portfolio])
+    hierarchy_search.process.field
+  end
+
+  def resolved_workflow_refs
+    [item_workflow_ref]
```

Comment:
I wonder if `item_workflow_ref` should be private? It appears the the method required would be `resolved_workflow_refs`.  Maybe rename `item_workflow_ref` -> `resolve_workflow_refs` ?

Reply:
Thinking through this, I wonder if we replace service with a small association based lookup?
```
[PortfolioItem.last, PortfolioItem.last.portfolio].map(&:workflow_ref).compact
```
The order of the `workflow_ref` ids would be correct and we could add in the Organization later which would not be hard.